src/distributions/bates.ts

```diff
@@ -1,8 +1,9 @@
-import ow from 'ow'
+// import ow from 'ow'
 import { Random } from '../random'
+import { NumberValidator } from '../NumberValidator'
 
 export default (random: Random, n = 1) => {
-  ow(n, ow.number.integer.positive)
+  new NumberValidator(n).isInt().isPositive()
```

Comment:
I really like the number validator API; way simpler than `ow`.

one thing to avoid is using `new` with side-effects.

so here, i think doing something like `numberValidator(n).isint().isPositive()` would make more sense.

Reply:
Could this be achieved via a

```typescript 
export function numberValidator(n: number): NumberValidator {
  return new NumberValidator(n)
}
```

or did you have another pattern in mind?

Reply:
yep